tasks/_agent-linux-macos-shared.yml

```
@@ -18,36 +18,50 @@
   when: datadog_manage_config
   notify: "{{ agent_dd_notify_agent }}"
 
-- name: Register all checks directories present in datadog
+- name: Register all check configs present in datadog
   find:
     paths: "{{ agent_dd_config_dir }}/conf.d/"
     patterns:
-      - "*.d"
-    file_type: directory
-  register: agent_datadog_conf_directories
+      - "conf.yaml*"
+    recurse: true
+    depth: 2
+    file_type: file
+  register: agent_datadog_conf_paths
   when: datadog_manage_config and (datadog_disable_untracked_checks or datadog_disable_default_checks)
 
 - name: Delete checks not present in agent_datadog_tracked_checks
   file:
     path: "{{ agent_dd_config_dir }}/conf.d/{{ item }}.d/conf.yaml"
     state: absent
-  loop: "{{ agent_datadog_conf_directories.files | map(attribute='path') | list | map('basename') | list | map('regex_replace', '^(.*).d$', '\\1') | list }}"
+  loop: >-
+    {{
+      agent_datadog_conf_paths.files | map(attribute='path') | select('match', '.*\/conf\.yaml$') | list
+      | map('dirname') | map('basename') | list | map('regex_replace', '^(.*).d$', '\1') | list
+    }}
```

💭 How about something like
```suggestion
    {{ agent_datadog_conf_paths.files | map(attribute='path') | select('match', '.*\/conf\.yaml$') | list }}
```

And replacing `path` to be:
```yaml
    path: "{{ item }}"
```
(and similarly for the others)?

That should do the same and allow us to simplify this filter chain significantly.

---

Works for me. It does change the Ansible task output a bit since the value of item is different. But I agree this is a much cleaner option.

---

Just kidding, I remember when I didn't do it like that. `item` needs to be the check name so it can be used in the `when` statement

---

Good point, I had missed that. I'll merge this as is, thanks again.